esp-hal/src/systimer.rs

```diff
@@ -20,10 +20,19 @@
 //!
 //! ## Example
 //! ```no_run
-//! let peripherals = Peripherals::take();
+#![doc = include_str!("../doc_helper/before")]
+//! # use esp_hal::systimer::SystemTimer;
+//! # use fugit::ExtU32;
+//! let systimer = SystemTimer::new(peripherals.SYSTIMER);
 //!
-//! let syst = SystemTimer::new(peripherals.SYSTIMER);
-//! println!("SYSTIMER Current value = {}", SystemTimer::now());
```

Comment:
if we want to use `println!` in the examples we'd need to add it as a test- or dev-dependency - not sure if it's worth it